_includes/display-logo.html

```diff
@@ -0,0 +1,18 @@
+<!-- Workaround for correctly resolving SVG images served by GitHub -->
+{% assign itemLogo = include.itemLogo %}
+{% assign screenshots = include.screenshots %}
+{% assign classes = include.classes %}
+{% if itemLogo contains 'github' and itemLogo contains '.svg' %}
+  {% assign logo=itemLogo | append: '?sanitize=true' %}
+{% else %}
+  {% assign logo=itemLogo %}
+{% endif %}
+<!-- Show images: 
+1. Screenshot
+2. Logo
+3. Default logo
+4. If everything else fails, defaults to placeholder
+ -->
+  <div class="{{ classes }}"
+    style="background: url({{ screenshots | first | default: logo | default:'/assets/images/placeholder-software-riuso.jpg'}}), url('/assets/images/placeholder-software-riuso.jpg') no-repeat top center;">
```

Comment:
This is suboptimal. I'd like to include a bit of logic in order to pick the correct placeholder based on the existence or not of the iPA code. However, I cannot get it from the contents passed to this include, like for example this is the output of the "relatedSw" array:
```
{"id"=>"71dc97ddb83597db9cc8d28008ec0139578f9d72", "publiccode"=>{"categories"=>["fleet-management", "data-collection", "data-analytics", "data-visualization"], "description"=>{"en"=>{"features"=>["Web based frontend for public applicants.", "Web based backoffice frontend to check, navigate submitted applications.", "RESTful backend", "Publishes real-time statistics on the system usage."], "genericName"=>"geofleet", "localisedName"=>"geofleet", "screenshots"=>["https://raw.githubusercontent.com/vvfosprojects/vvfgeofleet/master/docs/frontend_screenshot.png"]}}, "isBasedOn"=>nil, "legal"=>{"repoOwner"=>"Corpo Nazionale dei Vigili del Fuoco"}, "name"=>"geofleet", "url"=>"https://github.com/vvfosprojects/vvfgeofleet.git"}} {"id"=>"2ed8e07469a5b9dcdb6ff83dd01c415b6a0cd4be", "publiccode"=>{"categories"=>["fleet-management", "data-collection", "data-analytics", "data-visualization"], "description"=>{"it"=>{"features"=>["Web based frontend for operators.", "RESTful backend.", "Call management.", "Emergency Teams dispatch.", "GIS based emergency management."], "genericName"=>"sovvf", "localisedName"=>"Sala Operativa VVF", "screenshots"=>["https://raw.githubusercontent.com/vvfosprojects/sovvf/master/doc/Cattura1.png"]}}, "isBasedOn"=>nil, "legal"=>{"repoOwner"=>"Corpo Nazionale dei Vigili del Fuoco"}, "name"=>"Sala Operativa VVF", "url"=>"https://github.com/vvfosprojects/sovvf"}} {"id"=>"48367ef70aa3a70750a50383bbf460eeee5eddb6", "publiccode"=>{"categories"=>["employee-management"], "description"=>{"it"=>{"features"=>["Protocollazione automatizzata", "Form dinamici per creazione moduli di inserimento da backend", "calcolo punteggio automatico", "Cineca CSA - Integrazione mediante repliche dati (cache o sql)", "SSO SAML2"], "genericName"=>"applicazione web", "localisedName"=>"", "screenshots"=>["https://raw.githubusercontent.com/UniversitaDellaCalabria/PEO/master/data/gallery/home.png", "https://raw.githubusercontent.com/UniversitaDellaCalabria/PEO/master/data/gallery/domanda1.png"]}}, "isBasedOn"=>nil, "legal"=>{}, "name"=>"Procedura Elettronica Online", "url"=>"https://github.com/UniversitaDellaCalabria/PEO"}} {"id"=>"0ca54275dbb2d194a8c2ead4b44bdda4aead66ec", "publiccode"=>{"categories"=>["content-management", "website-builder"], "description"=>{"it"=>{"features"=>["editor HTML", "gestione SEO", "gestione modelli dei dati", "gestione menu di navigazione", "gestione utenti e ruoli", "backup dei dati"], "genericName"=>"Content Management System", "localisedName"=>"", "screenshots"=>["https://raw.githubusercontent.com/italia/agavecms/master/screenshots/admin.png", "https://raw.githubusercontent.com/italia/agavecms/master/screenshots/edit_content.png"]}}, "isBasedOn"=>nil, "legal"=>{"repoOwner"=>"Team per la Trasformazione Digitale - Presidenza del Consiglio dei Ministri"}, "name"=>"Agave CMS", "url"=>"https://github.com/italia/agavecms"}} 
```
Any clue @sebbalex? Thanks!

Reply:
Logic inserted. However, there's no way right now to get the `codiceIPA` from the `relatedSoftwares` section in `softwares.yml`. I believe we should add it here type software struct {
	ID         string `json:"id"`
	Slug       string `json:"slug"`
	PublicCode struct {
		URL         string   `json:"url"`
		Name        string   `json:"name"`
		IsBasedOn   []string `json:"isBasedOn"`
		Description map[string]struct {
			LocalisedName string   `json:"localisedName"`
			GenericName   string   `json:"genericName"`
			Features      []string `json:"features"`
			Screenshots   []string `json:"screenshots,omitempty"`
		} `json:"description"`
		Categories []string `json:"categories"`
		Legal      struct {
			RepoOwner string `json:"repoOwner,omitempty"`
		} `json:"legal,omitempty"`
	} `json:"publiccode"`


	// This is not populated from ES
	variants []software
}
or maybe we could query `softwares.yml` foreach entry in `relatedSoftwares`

Reply:
done in https://github.com/italia/developers-italia-backend/pull/154